ardere/aws.py

```diff
@@ -335,6 +432,14 @@ def shutdown_plan(self, steps):
         for page in response_iterator:
             service_arns.extend(page["serviceArns"])
 
+        # Avoid shutting down metrics if tear down was not requested
+        # We have to exclude it from the services discovered above if we
+        # should NOT tear it down
+        if not self._plan["influx_options"]["tear_down"]:
+            metric_service = self.locate_metrics_service()
```

Comment:
The metric service may already be running, so it wouldn't have been created at startup. Since shut-down needs to catch errors in other sections, service arn's or other info may be missing in the JSON blob passed around. So shut-down assumes nothing, and queries AWS to see what's there.